buildtools/cli.go

```diff
@@ -929,6 +941,52 @@ func NpmPublishCmd(c *cli.Context) (err error) {
 	return
 }
 
+func setupCmd(c *cli.Context) (err error) {
+	var packageManager project.ProjectType
+	packageManagerStr := c.Args().Get(0)
+	// If the package manager was provided as an argument, validate it.
+	if packageManagerStr != "" {
+		packageManager = project.FromString(packageManagerStr)
+		if !setup.IsSupportedPackageManager(packageManager) {
+			return errorutils.CheckErrorf("The package manager %s is not supported", packageManagerStr)
+		}
+	} else {
+		// If the package manager wasn't provided as an argument, select it interactively.
+		packageManager, err = selectPackageManagerInteractively()
+		if err != nil {
+			return
+		}
+	}
+	setupCmd := setup.NewSetupCommand(packageManager)
+	artDetails, err := cliutils.CreateArtifactoryDetailsByFlags(c)
+	if err != nil {
+		return err
+	}
+	setupCmd.SetServerDetails(artDetails).SetRepoName(c.String("repo")).SetProjectKey(c.String(cliutils.Project))
+	return commands.Exec(setupCmd)
+}
+
+func selectPackageManagerInteractively() (selectedPackageManager project.ProjectType, err error) {
+	allSupportedPackageManagers := setup.GetSupportedPackageManagersList()
+	var selected string
+	var selectableItems []ioutils.PromptItem
+	for _, packageManager := range allSupportedPackageManagers {
```

Comment:
Can inline the variable
```suggestion
	var selected string
	var selectableItems []ioutils.PromptItem
	for _, packageManager := range setup.GetSupportedPackageManagersList() {
```